.github/workflows/pull-request.yml

```diff
@@ -15,24 +15,47 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Download repository
-      uses: actions/checkout@v2
-    
-    - name: Setup Java
-      uses: actions/setup-java@v2
-      with:
-        distribution: 'adopt'
-        java-version: '11'
-        
-    - name: Build with Gradle
-      run: bash ./gradlew build --stacktrace
-      
-    - name: Upload APK
-      uses: actions/upload-artifact@v2
-      with:
-        name: pslab-android-apk
-        path: |
-          app/build/outputs/apk/fdroid/debug/app-fdroid-debug.apk
-          app/build/outputs/apk/fdroid/release/app-fdroid-release-unsigned.apk
-          app/build/outputs/apk/playstore/debug/app-playstore-debug.apk
-          app/build/outputs/apk/playstore/release/app-playstore-release-unsigned.apk
+      - name: Download repository
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+
+      - name: Build with Gradle
+        run: bash ./gradlew build --stacktrace
+
+      - name: Upload APK
+        uses: actions/upload-artifact@v2
+        with:
+          name: pslab-android-apk
+          path: |
+            app/build/outputs/apk/debug/app-debug.apk
+            app/build/outputs/apk/release/app-release-unsigned.apk
+
+      - name: Download Artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: pslab-android-apk
+
+      - name: Display structure of downloaded files
+        run: ls -R
+
+      - name: Add APK files to apk branch
```

Comment:
Try to incorporate this existing script. (You’re free to edit travis parts to match with GitHub actions)
https://github.com/fossasia/pslab-android/blob/development/scripts/upload-apk.sh

We’re only building apk for merge into master or releases.

Reply:
I have tried to incorporate the same script. I've also added a check so that .apk and .aab files would be generated in the apk branch only after we push to master.
P.S.- I'll squash the commits positively from now on.